docs: Mention the python-unversioned-command package on Fedora.

rakuco opened the pull request:
Fedora does not ship a `/usr/bin/python` by default anymore, but does have a
separate package that provides a `python` symlink to the `python3` binary.
While here, also mention a similar package that is required in Ubuntu Focal
and later, and clarify that Python 3 is no longer unsupported.

Fixes #20035.

rakuco commented:
@foolip earlier in the same file, there's a paragraph that says _Note that Python 2.7 is required. Using Python 3 is not supported_. Does that still officially apply? I've been running `wpt` with Python 3 here with no issues.

clopez commented on docs/running-tests/from-local-system.md:
````diff
@@ -22,6 +22,10 @@ On Debian or Ubuntu:
 sudo apt-get install python python-pip virtualenv
 ```
 
+On other Linux distros, it is important to have a package that provides a
+`python` binary. On Fedora, for example, that means installing the
+`python-unversioned-command` package.
````
It looks good. You can add this: On Ubuntu, it means installing the package ```python-is-python3```

rakuco replied:
Done!

foolip commented:
No, that's outdated documentation. Python 3 mostly works now and will become the default pretty soon now.

@Hexcles @stephenmcgruer can either of you update the docs?

Commit:
docs: Mention the python-unversioned-command package on Fedora.

Fedora does not ship a `/usr/bin/python` by default anymore, but does have a
separate package that provides a `python` symlink to the `python3` binary.
While here, also mention a similar package that is required in Ubuntu Focal
and later, and clarify that Python 3 is no longer unsupported.

Fixes #20035.

Hexcles reviewed (commented):
Maybe let's just update it in this PR, since the new instruction would in fact give users Python 3 on Fedora and Ubuntu.

> Note that we are still in the process of migrating from Python 2 to 3. Most things now work in Python 3.6+; if you run into problems, please file an issue with the "python3" label and/or try Python 2.7.

rakuco commented:
> Maybe let's just update it in this PR, since the new instruction would in fact give users Python 3 on Fedora and Ubuntu.
> 
> > Note that we are still in the process of migrating from Python 2 to 3. Most things now work in Python 3.6+; if you run into problems, please file an issue with the "python3" label and/or try Python 2.7.

There are some Python2-specific paths in the macOS and Windows in the same file too. Do you know if it's enough to replace them with some Python3 one? Which version needs to be used?

Hexcles commented:
> There are some Python2-specific paths in the macOS and Windows in the same file too. Do you know if it's enough to replace them with some Python3 one? Which version needs to be used?

Hmm that's a bit tricky. I'm not exactly sure which macOS versions have Python 3.6+ by default; neither have I tested py3 instructions on Windows (I only know if you have a working Python 3 environment it'd work).

Let's keep the Windows and macOS instructions for now, especially since both of their Python 2 instructions still work for now.

sideshowbarker commented:
> I'm not exactly sure which macOS versions have Python 3.6+ by default;

I don’t know well about any versions older than the latest macOS. But in my macOS 10.15.6 with XCode 11.6 —
```
$ ls /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions
total 0
drwxr-xr-x  10 root   320B May 28 14:11 3.7
lrwxr-xr-x   1 root     3B May 28 14:10 Current -> 3.7
```
And as far as older macOS versions, I can attest to Python 3.7 being the XCode-installed version since at least April (so, since whatever versions of macOS and XCode were the latest in April).

Also, I have no idea about what Python version (if any) that macOS might have installed in the case where someone doesn’t also have XCode installed. As far as I understand, that `/Library/Developer/CommandLineTools/Library/Frameworks/` stuff is installed by Python. So maybe if you don’t have XCode installed, you won’t have *any* Python installed at all.

rakuco commented:
I've pushed a new version of this change that mentions Python 3 is supported and preferred, while not touching the Windows and macOS sections.
